.github/workflows/rust.yml

```diff
@@ -26,8 +26,54 @@ jobs:
 
   integration:
     runs-on: ubuntu-18.04
-    container:
-      image: docker://rust:1.37.0-buster
     steps:
     - uses: actions/checkout@v1
-    - run: make test-integration
+
+    # Forks run in GitHub actions, as they don't have access to secrets.
+    - name: Test (Fork)
+      if: github.event.pull_request.head.repo.fork
+      run: make test-integration
+
+    # Create a build image on a Linkerd build host.
+    - name: Setup (Origin)
+      if: '!github.event.pull_request.head.repo.fork'
+      env:
+        DOCKER_HOST: "ssh://linkerd-docker"
+      run: |
+        mkdir -p ~/.ssh
+        # Create an identity file and protect before writing contents to it.
+        touch ~/.ssh/id && chmod 600 ~/.ssh/id
+        echo "${{ secrets.DOCKER_PRIVATE_KEY }}" >~/.ssh/id
+        # Use well-known public keys for the host to prevent middlemen.
+        echo "${{ secrets.DOCKER_KNOWN_HOSTS }}" >~/.ssh/known_hosts
+        # Configure host with BatchMode to prevent idlelocking.
+        (
+          echo "Host linkerd-docker"
+          echo "    User github"
+          echo "    Hostname ${{ secrets.DOCKER_ADDRESS }}"
+          echo "    IdentityFile ~/.ssh/id"
+          echo "    BatchMode yes"
+          echo "    ServerAliveInterval 60"
```

Comment:
Nit: Is 60 a little often? I'm seeing when set this can be something like 5 minutes. Also looks like a `ServerAliveCountMax` pairs well with this if we care about checking aliveness.

Reply:
```
     ServerAliveInterval
             Sets a timeout interval in seconds after which if no data has been received
             from the server, ssh(1) will send a message through the encrypted channel
             to request a response from the server.  The default is 0, indicating that
             these messages will not be sent to the server.
```

I've set this at 60s to ensure that we're sending messages on the connection even when a command being run stops emitting output (like the build does while building the proxy). WIthout this, the client can lose track of the server such that, even when the remote work completes, the client is never notified. Adding these heartbeats prevents that behavior. We can add `ServerAliveCountMax` if we run into situations where the network or build host dies and the client does not detect it, but that seems less of a concern.